### PR TITLE
 DBEX: add includeToxicExposure indicator to formData

### PR DESCRIPTION
**Pull request by freeheeling**
- Follow-on to https://github.com/department-of-veterans-affairs/vets-api/pull/16693
- Revert some additions in https://github.com/department-of-veterans-affairs/vets-api/pull/16802

## Summary

- *This work is behind a feature toggle (flipper): **YES**/~NO~*: `disability_526_toxic_exposure`
- Evaluate toxic exposure Flipper set for current user, assigning boolean value of a new `includeToxicExposure` attribute as part of the initial `formData` generated for a newly-created In Progress Form
  - replaces previous PR that included the same attribute as part of an In Progress Form's metadata
  - to be utilized on the frontend in tracking and reporting the inclusion of the toxic exposure section for the form
- Translate `"includeToxicExposure"` of input form to output form, conditionally determining the inclusion of the `toxicExposure` section to the translated form
- Add `"includeToxicExposure"` to `all_claims.json` test file, and confirm conditional existence of toxic exposure component in form creation
- Remove dead code – `translate_form4142(form_content)` method
- **Responsible team:** Disability Benefits Experience Team 2 (dBeX Carbs 🥖)

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/83847
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/81830
- https://github.com/department-of-veterans-affairs/vets-api/pull/16689
- https://github.com/department-of-veterans-affairs/vets-api/pull/16693
- https://github.com/department-of-veterans-affairs/vets-api/pull/16802

## Testing done

- unit test enablement of toxic exposure feature flag
- local data manipulated to confirm expected results
- plan to target staging users for additional verification

## What areas of the site does it impact?

- Form 526EZ

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [X]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [ ]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

Any suggestions concerning test improvements or additions welcomed.

### CODE REVIEW

File: app/controllers/v0/disability_compensation_forms_controller.rb [hunk 1]
```diff
@@ -105,6 +104,10 @@ def auth_rating_info
       authorize(api, :rating_info_access?)
     end
 
+    def form_content
+      @form_content ||= JSON.parse(request.body.string)
```
Comment (tblackwe) [inline app/controllers/v0/disability_compensation_forms_controller.rb:108]: The form content already exists on the SavedClaim object (`saved_claim.form_data`) with a convenience method `saved_claim.parsed_form` 
I'd prefer to reuse existing data unless there is a reason not to.

File: lib/evss/disability_compensation_form/data_translation_all_claim.rb [hunk 2]
```diff
@@ -705,6 +706,10 @@ def add_toxic_exposure
         { 'toxicExposure' => input_form['toxicExposure'] }
       end
 
+      def include_toxic_exposure
```
Comment (tblackwe) [inline lib/evss/disability_compensation_form/data_translation_all_claim.rb:709]: Where is this method being used?

File: app/controllers/v0/disability_compensation_forms_controller.rb (hunk 1 above)
Comment (aurora-a-k-a-lightning) [reply app/controllers/v0/disability_compensation_forms_controller.rb:108]: @tblackwe this method is being used before the `SavedClaim` is created. 
this method is used to save the data on line 57!
and then the memoized `@form_content` is used in the `create_submission` -> `includes_toxic_exposure?` methods later.

Comment (tblackwe) [reply app/controllers/v0/disability_compensation_forms_controller.rb:108]: I agree it is currently being called before savedclaim is created, but the data is not used until later, and in the `create_submission` method where we have already passed in the saved claim object

Comment (tblackwe) [reply app/controllers/v0/disability_compensation_forms_controller.rb:108]: re-reading Aurora's comment and getting myself confused.

If ya'll think the code is fine as is, I dont feel strongly. But I do think we are repeating ourselves by parsing the form from 2 different places.

File: lib/evss/disability_compensation_form/data_translation_all_claim.rb (hunk 2 above)
Comment (freeheeling) [reply lib/evss/disability_compensation_form/data_translation_all_claim.rb]: Removed in [fb2fd3d](https://github.com/department-of-veterans-affairs/vets-api/pull/16899/commits/fb2fd3d086be9c2e39d986d746c5df6902ca45a3). Thanks.

File: app/controllers/v0/disability_compensation_forms_controller.rb (hunk 1 above)
Comment (freeheeling) [reply app/controllers/v0/disability_compensation_forms_controller.rb:108]: I think there's a differentiation between a saved claim's form data being parsed and the request body's content, which is necessary to crated the saved claim. Memoizing, the data is parsed once.

The suggested refactor is viable, although slightly less readable and would require a second parsing. Also, the memoized instance variable, and its cached result, is already available.

Comment (tblackwe) [reply app/controllers/v0/disability_compensation_forms_controller.rb:108]: I'm good with the explanation, and happy for you to leave it as you have it. Thanks for taking a second look

**Review by tblackwe — APPROVED** [review]

**Pull request merged**